fusionauth/resource_fusionauth_generic_connector.go

```diff
@@ -0,0 +1,250 @@
+			},
+			"debug": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Determines if debug should be enabled to create an event log to assist in debugging integration errors.",
+			},
+			"headers": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "An object that can hold HTTPHeader key and value pairs.",
+			},
+			"http_authentication_password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				Description: "The basic authentication password to use for requests to the Connector.",
+			},
+			"http_authentication_username": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				Description: "The basic authentication username to use for requests to the Connector.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique Connector name.",
+			},
+			"read_timeout": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "The read timeout for the HTTP connection, in milliseconds. Value must be greater than 0.",
+			},
+			"ssl_certificate_key_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The Id of an existing Key. The X509 certificate is used for client certificate authentication in requests to the Connector.",
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func buildGenericConnector(data *schema.ResourceData) fusionauth.GenericConnectorConfiguration {
+	connector := fusionauth.GenericConnectorConfiguration{
+		AuthenticationURL: data.Get("authentication_url").(string),
+		BaseConnectorConfiguration: fusionauth.BaseConnectorConfiguration{
+			Id:    data.Get("id").(string),
+			Data:  data.Get("data").(map[string]interface{}),
+			Debug: data.Get("debug").(bool),
+			Name:  data.Get("name").(string),
+			Type:  fusionauth.ConnectorType_Generic,
+		},
+		ConnectTimeout:             data.Get("connect_timeout").(int),
+		HttpAuthenticationPassword: data.Get("http_authentication_password").(string),
+		HttpAuthenticationUsername: data.Get("http_authentication_username").(string),
+		ReadTimeout:                data.Get("read_timeout").(int),
+		SslCertificateKeyId:        data.Get("ssl_certificate_key_id").(string),
+	}
+
+	if i, ok := data.GetOk("headers"); ok {
+		connector.Headers = intMapToStringMap(i.(map[string]interface{}))
+	}
+
+	return connector
+}
+
+func createGenericConnector(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+	connector := buildGenericConnector(data)
+	resp, faErrs, err := CreateConnector(*client.FAClient, connector.Id, GenericConnectorRequest{Connector: connector})
+	if err != nil {
+		return diag.Errorf("CreateWebhook err: %v", err)
```

Comment:
```
return diag.Errorf("CreateConnector err: %v", err)
```

Reply:
@melm73 Thank you for this! In addition to the change above, there are a few linter warnings

Reply:
Thanks! should be fixed now.